Refactoring authentication

TyJacalan opened the pull request:
### Changelog
- New auth route /admin/sign_in
- Admins can now only sign in from the admin sign in route
- Fix uncentered auth header
- Add logo to auth pages

### Problem
Users have a share a single model, where admins and standard users are delineated based on a role (enum). Via Devise, users share authentication controllers where admins and standard users sign in from the same route.

### Solution
Create a new route and controller, which extends from the Devise Sessions Controller. Handle the behaviors of the users/sessions controller and admins/sessions controller to only accept users with the role _standard_ and _admin_ respectively. In the views, render the form urls conditionally based on the request path.

TyJacalan committed:
feat: create separate route for admins to sign in

- create admin/sessions controller
- modify users/sessions controller to only accept 'standard' users
- create admin/sign_in route
- modify new sessions form to conditionally render the url based on the
  request path (either admin or users)

TyJacalan committed:
style: center auth header text and add logo

cimafrancamae reviewed (changes requested):
Hiya, if I login on either the user or admin page using an unregistered email, I get an error 
`undefined method `role' for nil`

This appears on SessionsController#create

TyJacalan committed:
fix: check if current_user exists before checking for user role

when a user signs in with a unregistered email, an undefined role error
will appear because the edited session controllers checks for the
current user's role before checking if a current user exists.

TyJacalan commented:
> Hiya, if I login on either the user or admin page using an unregistered email, I get an error `undefined method `role' for nil`
> 
> This appears on SessionsController#create

Nice catch! I added a commit to address this bug.

The issue was that the custom session controllers check for the current user's role:
```
 if current_user.role != 'admin'
```

This meant that, clicking 'log in' will always check for the user's role even if the user doesn't exist. Since unregistered accounts do not have a role, it will render a error that the role is undefined.

The solution was to check first if the account is registered exists:
```
 if current_user && current_user.role && current_user.role != 'admin'
```

cimafrancamae reviewed (approved):
💯
